samcli/lib/build/build_graph.py

```diff
@@ -625,7 +627,12 @@ def get_build_dir(self, artifact_root_dir: str) -> str:
         Return the directory path relative to root build directory
         """
         self._validate_functions()
-        return self.functions[0].get_build_dir(artifact_root_dir)
+        build_dir = self.functions[0].get_build_dir(artifact_root_dir)
+        if is_experimental_enabled(ExperimentalFlag.BuildPerformance) and len(self.functions) > 1:
```

Comment:
Add message to customers about this change and add a link to a doc about all the changes that this includes.